governance/proposals/udt/020-swap-arb-bridge-to-mainnet.js

```diff
@@ -0,0 +1,318 @@
+  const arbBalance = await arbToken.balanceOf(L2_TIMELOCK_ALIAS)
+  const arbDecimals = await arbToken.decimals()
+
+  // Get the L2 Gateway Router
+  const gatewayRouterAddress = l2Network.tokenBridge.childGatewayRouter
+  const gatewayRouter = new ethers.Contract(
+    gatewayRouterAddress,
+    GATEWAY_ROUTER_ABI,
+    l2Provider
+  )
+
+  // Create interfaces
+  const bridge = new ethers.Contract(
+    UNLOCK_DAO_BRIDGE_ADDRESS,
+    BRIDGE_ABI,
+    l2Provider
+  )
+  const timelockAddress = await bridge.L1_TIMELOCK()
+
+  if (timelockAddress !== L1_DAO_TIMELOCK_ADDRESS) {
+    throw new Error('Timelock address is not correct')
+  }
+
+  // Get L2 UDT token address and balance
+  const l2UdtAddress =
+    await gatewayRouter.calculateL2TokenAddress(L1_UDT_ADDRESS)
+  const l2UdtToken = await getERC20Contract(l2UdtAddress, l2Provider)
+  const udtBalance = await l2UdtToken.balanceOf(L2_TIMELOCK_ALIAS)
+
+  // 1. transfer ARB tokens to the bridge contract
+  const transferArbCall = await createArbBridgeTicket({
+    from: timelockAddress,
+    to: L2_ARB_TOKEN_ADDRESS,
+    data: arbToken.interface.encodeFunctionData('transfer', [
+      UNLOCK_DAO_BRIDGE_ADDRESS,
+      arbBalance,
+    ]),
+    l1Provider,
+    l2Provider,
+  })
+
+  // 2. transfer ARB tokens to the bridge contract
+  const transferUdtCall = await createArbBridgeTicket({
+    from: timelockAddress,
+    to: l2UdtAddress,
+    data: arbToken.interface.encodeFunctionData('transfer', [
+      UNLOCK_DAO_BRIDGE_ADDRESS,
+      udtBalance,
+    ]),
+    l1Provider,
+    l2Provider,
+  })
+
+  // call contract to swap and bridge arb tokens and bridge UDT
+  const multicallCalls = [
+    {
+      contractAddress: UNLOCK_DAO_BRIDGE_ADDRESS,
+      calldata: bridge.interface.encodeFunctionData('swapAndBridgeArb'),
+    },
+    {
+      contractAddress: UNLOCK_DAO_BRIDGE_ADDRESS,
+      calldata: bridge.interface.encodeFunctionData('bridgeUdt'),
+    },
+  ]
+
+  // Encode the multicall data
+  const { data: multicallData, to: multicallAddress } =
+    await parseSafeMulticall({
+      calls: multicallCalls,
+      chainId: arbitrum.id,
+    })
+
+  // NB: this will fail because the state on Arbitrum is not ready (no ARB or UDT in bridge contract)
+  // Need to fund the contract on chain for this to work
+  const customBridgeCalls = await createArbBridgeTicket({
+    from: timelockAddress,
+    to: multicallAddress,
+    data: multicallData,
+    l1Provider,
+    l2Provider,
+  })
+
+  // Return the proposal
+  const proposalName = `# Bring back ARB and UDT on Arbitrum to Mainnet 
+
+This proposal goal is to bring back ARB tokens and UDT on Arbitrum to mainnet. To do this it 1) swaps ARB for ETH 2) bridges ETH and UDT back to the timelock on mainnet.
+
+To rpevent splitting the tasks into multiple proposals, it relies on the \`UnlockDAOArbitrumBridge\` contract deployed on Arbitrum at the address ${UNLOCK_DAO_BRIDGE_ADDRESS}.
```

Comment:
```suggestion
To prevent splitting the tasks into multiple proposals, it relies on the \`UnlockDAOArbitrumBridge\` contract deployed on Arbitrum at the address ${UNLOCK_DAO_BRIDGE_ADDRESS}.
```